src/plugins/plot/configuration/YAxisModel.js

```diff
@@ -280,7 +315,7 @@ export default class YAxisModel extends Model {
 
         this.set('values', yMetadata.values);
         if (!label) {
-            const labelName = seriesCollection
+            const labelName = seriesForThisYAxis
```

Comment:
I think we should extract this logic to a separate function and add some docs.
Also, on line 353 (not a change in this PR so I can't comment on the line), we do:

```js
this.set('label', labelUnits);
```

Is that correct?

Reply:
Yes, it is correct. The label is being used to either display the name or the units of the yAxis. I'm not sure why we're not showing both. That's probably a decision made by Charles.

Reply:
Refactored the code as much as possible with comments.

Reply:
Awesome. I can see it would make sense if the intention is to prefer labelUnits as the label if both a label name and label units are available.